src/web/core/src/test/java/org/geoserver/web/wicket/GeoServerDataProviderTest.java

```diff
@@ -0,0 +1,149 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.web.wicket;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+import java.util.Iterator;
+import java.util.List;
+import org.geotools.api.filter.Filter;
+import org.geotools.ows.wms.Layer;
+import org.junit.Test;
+
+public class GeoServerDataProviderTest {
+
+    private final GeoServerDataProvider<Object> geoServerDataProvider =
+            new GeoServerDataProvider<>() {
+                @Override
+                protected List<Property<Object>> getProperties() {
+                    return List.of(new BeanProperty<>("title", "title"));
+                }
+
+                @Override
+                protected List<Object> getItems() {
+                    return List.of(new Layer("key"), new Layer("keyword"));
+                }
+            };
+
+    @Test
+    public void fullTextFilter() {
+        geoServerDataProvider.setKeywords(new String[] {"keyword"});
+
+        Filter filter = geoServerDataProvider.getFilter();
+
+        assertEquals("[ AnyText is like *keyword* ]", filter.toString());
```

Comment:
Testing against toString is unreliable, developers are free to change its result without changing the semantics of the method (any string that "textually represents the object" is valid). Instead, use ``ECQL.toCQL(filter)``, it provides a text output with a format and stable grammar.

I'm not commenting the other cases in this class, but please fix them all.

Reply:
Fixed, thanks.
Removed also other `toString()` in assertions.